### PR TITLE
Enable IDE0020 (Use pattern matching)

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs [hunk 1]
```diff
@@ -3848,24 +3847,22 @@ private void WriteElements(string source, string? enumSource, ElementAccessor[]
...
         private void WriteText(string source, TextAccessor text)
         {
-            if (text.Mapping is PrimitiveMapping)
+            if (text.Mapping is PrimitiveMapping primitive)
             {
-                PrimitiveMapping mapping = (PrimitiveMapping)text.Mapping;
                 Writer.Write("WriteValue(");
                 if (text.Mapping is EnumMapping)
                 {
                     WriteEnumValue((EnumMapping)text.Mapping, source);
```
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs:3855]:
```suggestion
                if (text.Mapping is EnumMapping enum)
                {
                    WriteEnumValue(enum, source);
```

**Review by tannergooding — APPROVED** [review]
Nice to see so much code get more readable

**Review by eerhardt — APPROVED** [review]
LGTM

**Review by buyaa-n — APPROVED** [review]
Left one NIT, LGTM overall, thanks!

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs [hunk 2]
```diff
@@ -3848,24 +3847,22 @@ private void WriteElements(string source, string? enumSource, ElementAccessor[]
 
         private void WriteText(string source, TextAccessor text)
         {
-            if (text.Mapping is PrimitiveMapping)
+            if (text.Mapping is PrimitiveMapping primitive)
             {
-                PrimitiveMapping mapping = (PrimitiveMapping)text.Mapping;
                 Writer.Write("WriteValue(");
-                if (text.Mapping is EnumMapping)
+                if (text.Mapping is EnumMapping enum)
```
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs:3853]: Sorry wrong suggestion
```suggestion
                if (text.Mapping is EnumMapping enumMapping)
```

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs [hunk 3]
```diff
@@ -3848,24 +3847,22 @@ private void WriteElements(string source, string? enumSource, ElementAccessor[]
...
                 {
-                    WriteEnumValue((EnumMapping)text.Mapping, source);
+                    WriteEnumValue(enum, source);
```
Comment (buyaa-n) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs:3855]:
```suggestion
                    WriteEnumValue(enumMapping, source);
```

**Review by bartonjs — APPROVED** [review]

**Pull request merged**